### PR TITLE
dyalog: 19.0.48958 -> 19.0.49960, also apply patchelf to dyalog.rt

### LINKED ISSUE #316439: dyalog: isolates (parallel forEach and co) appear to be broken
### Describe the bug
Isolates parallelEach doesn't work

### Steps To Reproduce
run nix develop with
```nix
        devShells.default = with pkgs; mkShell rec {
          buildInputs = [
            (unfreePkgs.dyalog.override {
              acceptLicense = true;
            })
          ];
```
then run: dyalog
then in the dyalog repl type

```apl
)load isolate
⎕DL isolate.llEach (1 2 3)
```

### Expected behavior
⎕DL isolate.llEach (1 2 3) returning
```apl
1.00383 2.003841 3.004522
```
like its the case when using the .deb on lets say a debian host

### Additional context

strace log reveals a lot of lines like
```
clock_nanosleep(CLOCK_REALTIME, 0, {tv_sec=0, tv_nsec=10000000}, NULL) = 0
.... (hundreds of times)
clock_nanosleep(CLOCK_REALTIME, 0, {tv_sec=0, tv_nsec=10000000}, NULL) = 0

```
which does not happen on a debian host when I install dyalog 19 via dpkg -i

I have also tested this on a debian host that has nix installed and installed dyalog via nix instead of dpkg
but I run into the same issue making me believe there is something wrong with the drv

### Notify maintainers

@TomaSajt
@markus1189 

### Metadata

system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.18.2, nixpkgs: /etc/channels/nixpkgs

(I ran it against the current (1 hour ago) nixpkgs master/unstable branches) as in I pulled these 2 about 1 hour ago

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/316439

This PR bumps the version of the `dyalog` package. This is needed since Dyalog remvoed the previous version's files and since this is marked as unfree, it didn't get cached.

The original PR only addressed this:
Using the `isolate` workspace will create some internal `dyalog.rt` processes, but those weren't able to `dlopen` `libncurses.so` because I only `patchelf`-ed the main `dyalog` binary, causing it to crash.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/dy/dyalog/package.nix
Comment (SigmaSquadron) [inline pkgs/by-name/dy/dyalog/package.nix:150]:
```suggestion
    patchelf ${dyalogHome}/{dyalog,dyalog.rt} --add-needed libncurses.so
```
Removing the `for` loop makes the diff a bit more minimal. Up to you if you want to change it, as this somewhat decreases code legibility.

Comment (TomaSajt) [reply pkgs/by-name/dy/dyalog/package.nix:150]: I'll keep it like this, because some other binaries might be added to this list in the future.

**Review by SigmaSquadron — APPROVED** [review]

**Comment by SigmaSquadron** [comment]
For some reason OfBorg didn't notify the maintainers.

@TomaSajt @markus1189

**Comment by TomaSajt** [comment]
Apparently the previously used version's files were removed so I had to update them to the newest version.

**Comment by slotThe** [comment]
> Apparently the previously used version's files were removed so I had to update them to the newest version.

This also essentially means that `dyalog` can't be installed/changed until this lands

**Pull request merged**
